### DISCUSSION TITLE
to upload a text file such as TXT or PDF and summarize the text

### DISCUSSION THREAD

**Discussion by tmby1314** [langgenius/dify#5140] [Help]
I hope to upload a text file like TXT or PDF and summarize the text, rather than vectorizing and embedding the text into a knowledge base first, but I couldn't find the entry point.

**Comment by AndyMik90** [comment]
It would be a great feature.

**Comment by hponiang** [comment]
That is great，We need the feat

**Comment by wdzhwsh4067** [comment]
need it

**Comment by leicheng42** [comment]
me too

**Comment by GallantZhangYu** [comment]
need it

**Comment by fniu** [comment]
Please add this feature. I will be super happy.

**Comment by leicheng42** [comment]
on the way

**Comment by yajungu** [comment]
need it

**Comment by carloscesartc** [comment]
The addition of native support for PDF upload and analysis would make Dify a more robust and versatile solution. This would meet the demand of many companies and industries that need to process and analyze data directly from PDF documents, enabling AI agents to interact with real files and provide more precise and informed responses. The PDF upload and analysis feature would increase Dify's adoption and offer a key competitive advantage.

Thank you for your consideration!
